### ISSUE TITLE
`polygonize` should return a `MultiPolygon`, not a `Vector` of `Polygon`

### ISSUE THREAD

**Issue by rafaqz** [JuliaGeo/GeometryOps.jl#139]
See: https://discourse.julialang.org/t/polygonize-a-raster/114189/7

**Comment by asinghvi17** [comment]
Shouldn't it actually return a feature collection, with one feature for each unique value?  Or do you mean in the special case of an `AbstractMatrix{Bool}`, in which case I agree.

**Comment by rafaqz** [comment]
We don't actually do that case at the moment. But I'll add it, and yes that would need a `FeatureCollection`. 

Currently is just `Bool` and `f(x) -> Bool`. I guess we can allow `Integer` arrays to be converted to `FeatureCollection`? But floating point would need some more complicated contour function.

**Comment by asinghvi17** [comment]
That sounds good to me.

Makie uses Contour.jl and Isoband.jl to generate isolines - perhaps we could create an extension on that and use it as an algorithm if needed, the necessary code can be copied wholesale from Makie.

**Comment by rafaqz** [comment]
This is actually a very different algorithm, its exact pixels like the raster. But we could have those too.

**Comment by rafaqz** [comment]
I added handling for `Integer`, just returning a `FeatureCollection` of `MultiPolygon`. But maybe we can do this for other `Number` using either N isolines specified with a keyword `nintervals` (any better name ideas?) or with IntervalSets.jl intervals? Would that dependency be ok?

**Comment by rafaqz** [comment]
I have no guage of this being fast or not, its 10 million pixels and 400 categories...
```julia
julia> @time featurecollection = polygonize(rand(1:400, 10000, 1000));
 16.385245 seconds (29.87 M allocations: 7.798 GiB, 29.17% gc time)
```

**Comment by asinghvi17** [comment]
Maybe test against Contours.jl?  I can send a script later tonight once I get on the plane and have to stay awake

**Comment by asinghvi17** [comment]
Just thought of this but as you mentioned it's not really a comparison - the algorithms are too different!

**Comment by asinghvi17** [comment]
fixed by #141 I think
